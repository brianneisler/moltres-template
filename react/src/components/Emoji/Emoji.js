import { noop } from 'moltres-utils'
import emoji from 'node-emoji'
import PropTypes from 'prop-types'
import React from 'react'
import { StyleSheet, Text } from 'react-native'
import { compose, defaultProps, setPropTypes } from 'recompose'

const enhance = compose(
  setPropTypes({
    name: PropTypes.string.isRequired,
    onLongPress: PropTypes.func,
    onPress: PropTypes.func,
    style: Text.propTypes.style
  }),
  defaultProps({
    onLongPress: noop,
    onPress: noop,
    styles: StyleSheet.create({
      container: {
        flex: 1,
        justifyContent: 'center'
      }
    })
  })
)

export default enhance(({ name, ...rest }) => <Text {...rest}>{emoji.get(name)}</Text>)
