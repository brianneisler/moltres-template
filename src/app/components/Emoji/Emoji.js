export { Emoji as default } from 'emoji-mart'
