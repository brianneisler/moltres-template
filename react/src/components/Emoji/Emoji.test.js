import React from 'react'
import Enzyme, { render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Emoji from './Emoji'

Enzyme.configure({ adapter: new Adapter() })

it('converts basic emoji using sweat_smile', () => {
  const rendered = render(<Emoji name="sweat_smile" />)

  expect(rendered).toMatchSnapshot()
})

it('converts basic emoji using grinning', () => {
  const rendered = render(<Emoji name="grinning" />)

  expect(rendered).toMatchSnapshot()
})
