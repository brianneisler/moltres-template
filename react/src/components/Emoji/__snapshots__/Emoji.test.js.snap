// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts basic emoji using grinning 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-cursor-1loqt21 rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-focusable="true"
  dir="auto"
  styles="[object Object]"
  tabindex="0"
>
  ðŸ˜€
</div>
`;

exports[`converts basic emoji using sweat_smile 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-cursor-1loqt21 rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  data-focusable="true"
  dir="auto"
  styles="[object Object]"
  tabindex="0"
>
  ðŸ˜…
</div>
`;
